src/C4Network2IO.cpp

```diff
@@ -1248,6 +1264,43 @@ void C4Network2IO::SendConnPackets()
 		}
 }
 
+bool C4Network2IO::CheckPortAvailability(const uint16_t port)
+{
+	const auto pNetIO = std::make_unique<C4NetIOUDP>();
```

Comment:
You can't check whether a TCP port is available by attempting to bind a UDP socket.